src/pytest_plugins/forks/forks.py

```diff
@@ -125,12 +125,47 @@ class CovariantDescriptor:
     fork_attribute_name: str
     parameter_name: str
 
+    def get_marker(self, metafunc: Metafunc) -> pytest.Mark | None:
+        """
+        Get the marker for the given test function.
+        """
+        m = metafunc.definition.iter_markers(self.marker_name)
+        if m is None:
+            return None
+        marker_list = list(m)
+        assert len(marker_list) <= 1, f"Multiple markers {self.marker_name} found"
+        if len(marker_list) == 0:
+            return None
+        return marker_list[0]
+
     def check_enabled(self, metafunc: Metafunc) -> bool:
         """
         Check if the marker is enabled for the given test function.
         """
-        m = metafunc.definition.iter_markers(self.marker_name)
-        return m is not None and len(list(m)) > 0
+        return self.get_marker(metafunc) is not None
+
+    def filter_values(self, metafunc: Metafunc, values: List[Any]) -> List[Any]:
+        """
+        Filter the values for the covariant parameter.
+
+        I.e. if the marker has an argument, the argument is interpreted as a lambda function
+        that filters the values.
+        """
+        marker = self.get_marker(metafunc)
+        assert marker is not None
+        if len(marker.args) == 0:
+            return values
+        assert len(marker.args) == 1, f"Invalid number of arguments for marker {self.marker_name}"
+        filter_func = marker.args[0]
```

Comment:
```suggestion
        filter_func = marker.args[0]
        assert isinstance(filter_func, FunctionType), "First agument to marker must be of FunctionType"
```
If you want to add this check, don't forget:
```python
from types import FunctionType
```